daft/functions/ai/__init__.py

```diff
@@ -31,7 +27,7 @@
 ]
 
 
-def _resolve_provider(provider: str | Provider | None, default: str) -> Provider:
+def _resolve_provider(provider: ProviderType | Provider | None, default: ProviderType) -> Provider:
```

Comment:
Here the provider is just a string because its purpose is as an identifier for name resolution. You can implement your own providers with their own names.

```suggestion
def _resolve_provider(provider: str | Provider | None, default: ProviderType) -> Provider:
```